app/src/main/java/io/homeassistant/companion/android/vehicle/EntityGridVehicleScreen.kt

```diff
@@ -58,13 +71,21 @@ class EntityGridVehicleScreen(
         }
     }
 
-    override fun onGetTemplate(): Template {
+    fun getEntityGridItems(entities: List<Entity<*>>): ItemList.Builder {
         val manager = carContext.getCarService(ConstraintManager::class.java)
         val gridLimit = manager.getContentLimit(ConstraintManager.CONTENT_LIMIT_TYPE_GRID)
-
+        val shouldSwitchServers = serverManager.defaultServers.size > 1
+        val extraGrid = if (shouldSwitchServers) 3 else 2
         val listBuilder = ItemList.Builder()
+        val domainGrid = MainVehicleScreen(
```

Comment:
Always creating this grid even if it won't be used is wasteful. 

If I understand correctly, this is used as a 'starting grid' for when there are no entities, whereas if there are entities this function creates the starting grid itself (the forEachIndexed loop below). Why not create both of the starting grids outside the function to keep it simpler? Then you can do something like
```kotlin
val grid = if (isFavorites) entityGrid(entities) else domainGrid(domains)
if (isFavorites) {
    // Add navigation
    // Add all entities
    // Add change server
}
return grid
```

Reply:
thanks your understanding is correct and this is simpler. Thank you